submit_cgap/tests/test_submission.py

```diff
@@ -2202,3 +2253,144 @@ def test_running_on_windows_native():
         os_name, is_windows = pair
         with mock.patch.object(os, "name", os_name):
             assert running_on_windows_native() is is_windows
+
+
+@pytest.mark.parametrize(
+    "directory,file_name,recursive,glob_results,expected_file_path,expected_msg",
+    [
+        ("foo", "bar", False, [], "foo/bar", False),
+        ("foo", "bar", True, [], "foo/bar", False),
+        ("foo", "bar", False, ["foo/bar"], "foo/bar", False),
+        ("foo", "bar", False, ["foo/bar", "fu/foo/bar"], None, True),
+    ]
+)
+def test_search_for_file(
+    directory, file_name, recursive, glob_results, expected_file_path, expected_msg
+):
+    """Test output file path +/- error message dependent on file search
+    via glob.
+    """
+    with mock.patch(
+        "submit_cgap.submission.glob.glob", return_value=glob_results
+    ) as mocked_glob:
```

Comment:
In general, I'd ask you not to use `mock.patch` at all in SubmitCGAP. Only use `mock.patch.object`. There are places where that means you need to import the module itself as the object to bind, but you can do that with a relative import in this library because the `tests` subdir is a child of `submit_cgap`. (You're forced to use `mock.patch` on the actual library name in repos that have the `tests` dir be a sibling of the library or app dir. But I strongly prefer relative naming where possible to avoid ambiguities and allow for renaming.

Some of that is overkill here, though. You could just `import glob` in this case and then do (untested):
```suggestion
    with mock.patch.object(glob, "glob", return_value=glob_results) as mocked_glob:
```